packages/in_app_purchase/in_app_purchase_android/android/src/main/java/io/flutter/plugins/inapppurchase/Translator.java

```diff
@@ -233,6 +235,34 @@ static HashMap<String, Object> fromBillingResult(BillingResult billingResult) {
     return info;
   }
 
+  static HashMap<String, Object> fromUserChoiceDetails(UserChoiceDetails userChoiceDetails) {
+    HashMap<String, Object> info = new HashMap<>();
+    info.put("externalTransactionToken", userChoiceDetails.getExternalTransactionToken());
+    info.put("originalExternalTransactionId", userChoiceDetails.getOriginalExternalTransactionId());
+    info.put("productsList", fromProductsList(userChoiceDetails.getProducts()));
```

Comment:
The Dart object's field is just called `products`; is `List` added automagically by JSON serialization, or is this a mismatch?

Reply:
Confirmed mismatch and added test.